lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -684,6 +691,18 @@ private boolean checkTablePermission(ConnectorSecurityContext context, SchemaTab
         return false;
     }
 
+    private boolean checkTableFunctionPermission(ConnectorSecurityContext context, SchemaRoutineName function)
+    {
+        ConnectorIdentity identity = context.getIdentity();
+        for (TableFunctionAccessControlRule rule : tableFunctionRules) {
+            Optional<Boolean> allowed = rule.matches(identity.getUser(), identity.getEnabledSystemRoles(), identity.getGroups(), function);
```

Comment:
getEnabledSystemRoles or getEnabledRoles?

Reply:
In connector-based access control everything else refers to `getEnabledSystemRoles` not `getEnabledRoles`